src/Application/IndexAdapterInterface.php

```diff
@@ -16,7 +17,7 @@ public function delete(IndexConfigurationInterface $indexConfiguration): void;
 
     public function flush(string $index): void;
 
-    public function createNewWriteIndex(IndexConfigurationInterface $indexConfiguration): string;
+    public function createNewWriteIndex(AliasedIndexConfiguration $indexConfiguration): string;
```

Comment:
Just a question, why this change? Does this make something easier? I would rather have interfaces depend on other interfaces. Even more so because it already is dependent on the `IndexConfigurationInterface`

Reply:
gonna take a moment for me to get my mind back into it, but I'm pretty sure this change because this doesn't actually work with anything other than the `AliasedIndexConfiguration` it's expected in places further down the code, so allowing an interface, but needing the real class is weird, as it just needs the real class. but it could be put back, it would just be a vector for unknown errors

Reply:
I do not follow your reasoning. I know currently `createNewWriteIndex` is only called internally and only passed the class `AliasedIndexConfiguration`. But internally we do not rely on the fact that `AliasedIndexConfiguration` is passed. That is why we have the interface, that the contract we should (and do) adher to.  And which php will enforce for us do you strict typing. To me this isn't a vector of unknown errors. It is an a stricter contract between the code. I do also do not see why it should break with `IndexConfigurationInterface` as it has always worked any non of your changes breaks the original contract.

If you could explain more about the " vector for unknown errors" you see I could maybe lean your way